### DISCUSSION TITLE
zh/Q-and-A/answer/question6

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#919] [Announcements]
# zh/Q-and-A/answer/question6

我调试的时候发现一直卡在for循环里 问题描述： 为什么我的代码编译没有出错，但是输入后却一直输出不了结果？已经研究了两个小时了 image.png image.png

https://cygnomatic-controller.github.io/zh/Q-and-A/answer/question6.html

**Comment by become-the-most-dazzling** [comment]
问题
1.num没有初始化，而if语句中直接使用了该变量。
2.while中最后一条语句改变了i的值，可能会影响for的判断
解决方法
1.可以将int num ;改为int num = 0;令num初始化
2.for循环使用临时变量，防止while对for产生影响。
可以改为 for(i = a; i <= b; i++) {
        int temp = i;  // 使用临时变量保存 i 的值
        while(temp > 0) {
            c = temp % 10;
            if(c == 2) {
                num++;
            }
            temp = temp / 10;

**Reply by become-the-most-dazzling** [reply]
“}”   最后的大括号忘加了
